pkg/kubeadm/master/addons.go

```diff
@@ -223,11 +230,15 @@ func createKubeDNSServiceSpec(s *kubeadmapi.KubeadmConfig) (*api.ServiceSpec, er
 }
 
 func CreateEssentialAddons(s *kubeadmapi.KubeadmConfig, client *clientset.Clientset) error {
-	kubeProxyDaemonSet := NewDaemonSet(kubeProxy, createKubeProxyPodSpec(s))
-	SetMasterTaintTolerations(&kubeProxyDaemonSet.Spec.Template.ObjectMeta)
+	arches := [3]string{"amd64", "arm", "arm64"}
+
+	for _, arch := range arches {
```

Comment:
I mean, you'd have to do it for `kube-discovery` and `kube-dns` as well. I totally get how this works. I mean that we should have node arch labels, but get rid of the loop which creates it for all arches. One of the negative UX aspects is that users will see extra daemonsets and deployments they didn't ask for.